core/src/main/java/cucumber/api/TestCase.java

```diff
@@ -13,14 +13,16 @@
 public class TestCase {
     private final PickleEvent pickleEvent;
     private final List<TestStep> testSteps;
+    private final boolean dryRun;
 
-    public TestCase(List<TestStep> testSteps, PickleEvent pickleEvent) {
```

Comment:
Because this is an `api` package, should the ctor with the current signature be kept for backward compatibility? @mpkorstanje WDYT?

Reply:
Along with adding a deprecation tag that would be the correct course of of action.

In the long run (and I'm only just starting to think about this) I'd like TestCase to be an interface so we can exclude the constructors and run method from the public api. These methods aren't needed as part of the event bus protocol.

Reply:
Can we push this as a extra commit on this feature branch so we don't have to bother Adrian with this?

Reply:
I can't. Permissions have apparently not been set.

Reply:
I've pushed my suggested changes to https://github.com/cucumber/cucumber-jvm/tree/adrian-baker/master. I reckon we can finish up this discussion the PR for that branch. I've got no further objections to this branch being merged.

Reply:
Great, thanks. Yes wasn't comfortable with having to change an api package, but in the current state didn't seem avoidable.